internal/datanode/compaction/clustering_compactor.go

```diff
@@ -1013,16 +1068,28 @@ func (t *clusteringCompactionTask) scalarPlan(dict map[interface{}]int64) [][]in
 	return buckets
 }
 
-func (t *clusteringCompactionTask) refreshBufferWriter(buffer *ClusterBuffer) error {
-	segmentID, err := t.allocator.AllocOne()
-	if err != nil {
-		return err
+func (t *clusteringCompactionTask) refreshBufferWriter(buffer *ClusterBuffer) (bool, error) {
+	var segmentID int64
```

Comment:
Can we spilt out a new method `needPack`? seems a little mix-together now.

Reply:
ok